larreco/RecoAlg/TrackMomentumCalculator.cxx

```diff
@@ -38,37 +38,97 @@ using std::endl;
+       1.848, 1.771, 1.713, 1.67,  1.609, 1.57,  1.536, 1.519, 1.508, 1.51,  1.517, 1.526, 1.537,
+       1.548, 1.559, 1.57,  1.591, 1.61,  1.628, 1.645, 1.675, 1.7,   1.733, 1.761, 1.799, 1.829,
+       1.855, 1.877, 1.897, 1.914, 1.93,  1.944, 1.969, 1.991, 2.01,  2.028, 2.058, 2.084, 2.119,
+       2.149, 2.193, 2.232, 2.269, 2.304, 2.337, 2.369, 2.4,   2.43,  2.49,  2.548, 2.606, 2.663,
+       2.776, 2.888, 3.055, 3.224, 3.498, 3.774, 4.052, 4.332}};
+    for (double& value : dEdx) {
+      value *= LAr_density * 1e-3; // convert to GeV/cm
+    }
+    return dEdx;
+  }
+
+  const int ndedx = 73;
+  const std::vector<double> dEdx_GeV_per_cm = dedx_GeV_per_cm();
+  const std::vector<double> E_GeV{
+    {0.115695, 0.117697, 0.119693, 0.122694, 0.125695, 0.13069,  0.135694, 0.14069,  0.145714,
+     0.150689, 0.155682, 0.160666, 0.165693, 0.17566,  0.185714, 0.1957,   0.205644, 0.225683,
+     0.245698, 0.275669, 0.305658, 0.355669, 0.405711, 0.45563,  0.505671, 0.555646, 0.605694,
+     0.655676, 0.705661, 0.805664, 0.905689, 1.00557,  1.10606,  1.30529,  1.50571,  1.8061,
+     2.10565,  2.60614,  3.1058,   3.60555,  4.10536,  4.60521,  5.10609,  5.606,    6.10591,
+     7.10579,  8.10569,  9.10561,  10.1106,  12.1105,  14.1104,  17.1103,  20.1103,  25.1102,
+     30.1102,  35.1102,  40.1101,  45.1101,  50.1101,  55.1101,  60.1101,  70.1101,  80.1101,
+     90.1101,  100.1,    120.1,    140.1,    170.1,    200.1,    250.1,    300.1,    350.1,
+     400.1}};
+  TGraph const dEdx_vs_E{ndedx, &E_GeV[0], &dEdx_GeV_per_cm[0]};
+  TSpline3 const dEdx_vs_E_spline3{"dEdx_vs_E", &dEdx_vs_E};
+
   TVector3 const basex{1, 0, 0};
   TVector3 const basey{0, 1, 0};
   TVector3 const basez{0, 0, 1};
-  constexpr float kcal{0.0024}; // Approximation of dE/dx for mip muon in LAr
+  constexpr double kcal{0.0024}; // Approximation of dE/dx for mip muon in LAr
 
+  constexpr double MomentumDependentConstant(const double p)
+  {
+    // values measured with MC: https://indico.fnal.gov/event/62283/contributions/299163/attachments/181289/248600/20240910_SimReco_Update_CM.pdf
+    double a = 0.079;
+    double c = 10.435;
+    return (a / (p * p)) + c;
+  }
+  double ComputeExpetecteRMS(const double p, const double red_length)
+  {
+    double beta = std::sqrt(1 - ((m_muon * m_muon) / (p * p + m_muon * m_muon)));
+    return (MomentumDependentConstant(p) / (p * beta)) *
+           (1.0 + 0.038 * TMath::Log(red_length / cet::square(beta))) * std::sqrt(red_length);
+  }
   class FcnWrapper {
   public:
     explicit FcnWrapper(std::vector<double>&& xmeas,
                         std::vector<double>&& ymeas,
-                        std::vector<double>&& eymeas)
-      : xmeas_{xmeas}, ymeas_{ymeas}, eymeas_{eymeas}
+                        std::vector<double>&& eymeas,
+                        double correction)
+      : xmeas_{xmeas}, ymeas_{ymeas}, eymeas_{eymeas}, correction_{correction}
```

Comment:
In order to benefit from move semantics, even r-value references must be moved.  So all of the vectors should be initialized like:

```c++
  : xmeas_{std::move(xmeas)}, ymeas{std::move(ymeas)}, eymeas_{std::move(eymeas)}, correction_{correction}
```

You're just adding the `correction` variable to the constructor, but it looks like previous code authors missed the `std::move`.